playwright/_impl/_transport.py

```diff
@@ -34,10 +40,46 @@ def _get_stderr_fileno() -> Optional[int]:
         return sys.__stderr__.fileno()
 
 
-class Transport:
+class Transport(ABC):
+    def __init__(self) -> None:
+        self.on_message = lambda _: None
+
+    @abstractmethod
+    def stop(self) -> None:
+        pass
+
+    def cleanup(self) -> None:
+        pass
+
+    @abstractmethod
+    async def wait_until_stopped(self) -> None:
+        pass
+
+    async def run(self) -> None:
+        self._loop = asyncio.get_running_loop()
+        self.on_error_future: asyncio.Future[bool] = asyncio.Future()
```

Comment:
Is Future typed in 3.7?

Reply:
removed, since we have it never like that and in this case I only use it to throw an exception.